joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstCreator.scala

```diff
@@ -103,4 +115,70 @@ class AstCreator(
   protected def astForNode(json: Value): Seq[Ast] = {
     astForNode(createParserNodeInfo(json))
   }
+
+  def init(): ParserResult = {
+    val parserResult = GoAstJsonParser.readFile(Paths.get(jsonAstFilePath))
+    lineNumberMapping = positionLookupTables(parserResult)
+    declaredPackageName = parserResult.json(ParserKeys.Name)(ParserKeys.Name).str
+    fullyQualifiedPackage = goMod.getNameSpace(parserResult.fullPath, declaredPackageName)
+    fileName = parserResult.filename
+    originalFilePath = parserResult.fullPath
+    parserResult
+  }
+
+  def cacheSerializeAndStore(): Unit = {
+    tempAliasToNameSpaceMappingFilePath.map(file => {
+      Files.write(file, serialise(aliasToNameSpaceMapping))
+      aliasToNameSpaceMapping.clear()
+    })
+    lineNumberMapping = Map.empty
+  }
+
+  def loadCacheToProcess(): Unit = {
+    tempAliasToNameSpaceMappingFilePath.map(file => {
+      val deserialised = deserialise(Files.readAllBytes(file))
+      aliasToNameSpaceMapping = deserialised.asInstanceOf[mutable.Map[String, String]]
+    })
+  }
+
+  def cleanup(): Unit = {
+    methodAstParentStack.clear()
+    aliasToNameSpaceMapping.clear()
+    lineNumberMapping = Map.empty
+    parserNodeCache.clear()
+    tempAliasToNameSpaceMappingFilePath.map(file => {
+      if (Files.exists(file)) {
+        Files.delete(file)
+      }
+    })
+  }
+
+  /** Serialise any object to byte array to be passed through queue
+    *
+    * @param value
+    *   \- Any object to passed through queue as a result item.
+    * @return
+    *   \- Object serialised into ByteArray
+    */
+  private def serialise(value: Any): Array[Byte] = {
+    val stream: ByteArrayOutputStream = new ByteArrayOutputStream()
+    val oos                           = new ObjectOutputStream(stream)
+    oos.writeObject(value)
+    oos.close()
+    stream.toByteArray
+  }
+
+  /** Deserialize the ByteArray back to Object.
+    *
+    * @param bytes
+    *   \- Array[Byte] to be deserialized
+    * @return
+    *   \- Deserialized object
+    */
+  private def deserialise(bytes: Array[Byte]): Any = {
+    val ois   = new ObjectInputStream(new ByteArrayInputStream(bytes))
+    val value = ois.readObject
```

Comment:
`readObject` is an RCE waiting to happen, it is an extremely dangerous function.

We have a whole talk at BSides about exploiting Java deserialization https://www.youtube.com/watch?v=8qo0ZGK0tt0

https://msgpack.org is on the classpath from ODB, rather serialize data using that format.

Reply:
I guess serialising the alias to namespace mapping is anyway shouldn't make much of the difference. I need to try removing this.